### DISCUSSION TITLE
How to trigger actions re-run for given commit

### DISCUSSION THREAD

**Discussion by jeroen** [octokit/octokit.js#1820] [General]
I would like to automate something that I currently do manually via the web UI: re-run the my workflow for a given commit. 

In the UI I click on the green/red checkmark behind the commit, and then the "rerun all jobs" button. But I am running into two problems when doing this with the API:

 - The `GET /repos/:owner/:repo/commits/:ref/check-runs` api returns the check-run-id's but we need a `run_id`. There is an api to get recent runs, however I can't find a way to get the `run_id ` for a given commit. Is there a way to get the `run_id` for a given commit or `check-run` ?

- With `POST /repos/:owner/:repo/actions/runs/:run_id/rerun` you get a 403 permission denied error when trying to re-run a workflow that has previously succeeded? This is strange because this is allowed in the UI.

**Comment by gr2m** [comment]
Hi Jeroen, these are great questions. It would be great if you could ask them at https://github.community/c/github-api-development-and-support/37, as they are API related, and not a problem with Octokit. There are GitHub support folks their who will be better able to help then myself.

> returns the check-run-id's but we need a `run_id`

I thought they were the same? In your question on I would recommend to share example requests with response, ideally using curl
